include/zephyr/bluetooth/bluetooth.h

```diff
@@ -346,6 +346,20 @@ int bt_id_reset(uint8_t id, bt_addr_le_t *addr, uint8_t *irk);
  */
 int bt_id_delete(uint8_t id);
 
+/**
+ * @brief Bluetooth data header size.
+ *
+ * Size of the type and the length field of the Bluetooth data.
+ */
+#define BT_DATA_HDR_SIZE 2
+
+/**
+ * @brief Bluetooth data size.
+ *
+ * Get the size of a @ref bt_data given its data length.
+ */
+#define BT_DATA_SIZE(data_len) ((data_len) + BT_DATA_HDR_SIZE)
```

Comment:
Given how `struct bt_data` is defined, I'm worried that this may be misunderstood:

```c
struct bt_data {
	uint8_t type;
	uint8_t data_len;
	const uint8_t *data;
};
```

There is no `hdr` field in `bt_data`, and how it's defined how could incorrectly indicate that the first 2 octets of `data` is the `hdr`, as it is in most other BT APIs in Zephyr.

Reply:
The `BT_DATA_SIZE` macro basically does the same thing as `bt_data_get_len`, doesn't it? Why do we need both?

Reply:
> There is no `hdr` field in `bt_data`, and how it's defined how could incorrectly indicate that the first 2 octets of `data` is the `hdr`, as it is in most other BT APIs in Zephyr.

It's maybe not done like that in the other BT APIs in Zephyr but I think here it make sense. There is an explicit comment that define what the value is and the `data` field is clearly defined in the spec and doesn't have a header.

I think having that value is useful. But if you have a better naming, I'll take it :)

> The `BT_DATA_SIZE` macro basically does the same thing as `bt_data_get_len`, doesn't it? Why do we need both?

`bt_data_get_len` work on an an array of `bt_data` struct. Here we take only one `data_len` as input.

Reply:
> bt_data_get_len work on an an array of bt_data struct. Here we take only one data_len as input.

You can also just provide a single `bt_data` for that function to get the same functionality :) I don't see the need for this macro, and IMO it should be removed again.

Reply:
Do we need this in addition to `bt_data_get_len`?: Maybe. It's the typical macro vs function tradeoff: `bt_data_get_len` is not a 'constant expression', so it cannot be used in global initializers or type declarations. In those cases, a macro is more powerful. This PR does use this ability as far as I can see, so it might as well just use `bt_data_get_len`.

I think this macro is useful, but it should have different semantics; It's not the "the size of a `bt_data` given its data length", but "The output size of `bt_data_serialize`, determined by `bt_data.data_len`"[^0]. I would also rename it to "bt_data_serialized_size".

[^0]: Unnecessarily abstract alternatives I considered:
"The size of any 'AD Structure' represented by a `struct bt_data` with the given `data_len`."
"The size of any 'AD Structure' containing 'AD Data' of the given length."